### ISSUE TITLE
How to enumerate CKCollectionViewDataSource's index paths?

### ISSUE THREAD

**Issue by cfcommando** [facebook/componentkit#164]
CKComponentDataSource has a lot of useful methods that are totally inaccessible because it's internal to the collectionview datasource. Am I missing something?

Enumerating would be very useful externally for reloading the collectionview against a new set of models (the current reload method appears to be implemented internally by iterating and updating against the same models, instead of different ones).

**Comment by cfcommando** [comment]
To expand on this: The use case here is pull to refresh on fast-moving content where I can expect the incoming data to be disjoint from the original data.

`CKCollectionViewDataSource` hides a lot of interface. `- (void)updateContextAndEnqeueReload:(id)newContext;` doesn't allow for replacing the models, so deleting+re-inserting everything is necessary. Without being able to query the `CKCollectionViewDataSource`'s state, the currently-best way to do a total reload is cross my fingers and hope that the data source activity has quiesced and that its valid index paths are synchronized with the UICollectionView - then query the collection view to produce the index paths to delete.

Basically, being able to query the count or enumerate the index paths for a quiesced collection view, even if the views haven't appeared yet, would make enqueuing changes into the pipeline much less risky.

**Comment by benlodotcom** [comment]
Great remark, the fact that we don't expose the input state of the dataSource is intentional though. Computing a changeset (a reload or a regular one) requires to keep track of the state of your data externally.

http://componentkit.org/docs/datasource-gotchas.html#don-t-ask-the-the-list-view-for-the-position-of-the-next-insertion
